src/GovUk.Education.ExploreEducationStatistics.Content.Services/ReleaseService.cs

```diff
@@ -2,85 +2,133 @@
-            _methodologyCacheService = methodologyCacheService;
-            _publicationCacheService = publicationCacheService;
+            _releaseFileRepository = releaseFileRepository;
             _userService = userService;
+            _mapper = mapper;
         }
 
-        // TODO EES-3643 - move into a ReleaseCacheService?
-        public async Task<Either<ActionResult, ReleaseViewModel>> GetCachedViewModel(string publicationSlug, string? releaseSlug = null)
+        public async Task<Either<ActionResult, CachedReleaseViewModel>> GetRelease(string publicationSlug,
+            string? releaseSlug = null)
         {
-            return await _publicationCacheService.GetPublication(publicationSlug)
-                .OnSuccessCombineWith(publication => _methodologyCacheService.GetSummariesByPublication(publication.Id))
-                .OnSuccess(async tuple =>
+            return await _persistenceHelper.CheckEntityExists<Publication>(q =>
+                    q.Include(p => p.Releases)
+                        .Where(p => p.Slug == publicationSlug)
+                )
+                .OnSuccess(publication =>
                 {
-                    var (publication, methodologies) = tuple;
-                    return await GetCachedRelease(publicationSlug, releaseSlug)
-                        .OnSuccess(cachedRelease =>
-                        {
-                            var result = new ReleaseViewModel(
-                                cachedRelease!,
-                                publication
-                            )
-                            {
-                                Publication =
-                                {
-                                    Methodologies = methodologies
-                                }
-                            };
-
-                            return result;
-                        });
+                    // If no release is requested get the latest published version of the latest published release.
+                    // Otherwise get the latest published version of the requested release.
+                    return releaseSlug == null
+                        ? publication.LatestPublishedRelease()
+                        : publication.Releases
+                            .SingleOrDefault(r => r.Slug == releaseSlug && r.IsLatestPublishedVersionOfRelease());
+                }).OnSuccess(async release =>
+                {
+                    if (release == null)
+                    {
+                        return new NotFoundResult();
+                    }
+                    // Build the view model for the published release
+                    return await GetRelease(release.Id);
                 });
         }
 
-        public async Task<Either<ActionResult, ReleaseSummaryViewModel>> GetSummary(string publicationSlug,
-            string? releaseSlug)
+        public async Task<Either<ActionResult, CachedReleaseViewModel>> GetRelease(Guid releaseId,
```

Comment:
Thanks for taking the time to decipher what's going on!

Yes that's the case, it's preparing it for caching, but as mentioned in my previous PR this isn't a new class or a new name for it, it existed before this PR with the same name.

The problem seems to be we're caching resources but then transforming them before they are returned.

Properties only in cached view model `CachedReleaseViewModel`:
```
string DataGuidance
string PreReleaseAccessList
```

and properties only in response view model `ReleaseViewModel`:
```
bool HasPreReleaseAccessList
bool HasDataGuidance
bool LatestRelease
PublicationViewModel Publication
```

Some consumers in the Content.API want the cached release but not get the `ReleaseViewModel` version of it that has the publication which in turn has its methodology summaries . See `PreReleaseAccessListController` and `DataGuidanceService` which use the `CachedReleaseViewModel`.

To try and tidy some of this up and possibly abstract out everything related to publication away from getting a release, I created EES-3719.. However due to the differences between them I decided that even without the publication I woudn't be able to collapse `CachedReleaseViewModel` and `ReleaseViewModel` into one. Instead it just gets renamed `ReleaseCacheViewModel`.

As part of that work I realised `PublicationViewModel` has a similar issue but because there's only one of them it's a bit more difficult to spot the difference. The underlying non-cached service has a method `Get` which returns `PublicationViewModel` but never sets property `Methodologies` .

`PublicationViewModel.Methodologies` is only ever set after retrieving it from the cache, just before it's returned by a controller.

So to make it clearer what is cached and what's not, I created a `PublicationCacheViewModel` there too. This seems to 'legitimise' the need for two versions of the view model. See https://github.com/dfe-analytical-services/explore-education-statistics/pull/3509.

If there were no differences there wouldn't be two versions of the view models but I'm not sure if that's achievable without either returning everything where it might not be needed by the FE like the `DataGuidance` and  `PreReleaseAccessList`, or make the FE make multiple calls to get various bits of information.